--- hunspell-1.2.8.orig/tests/test.sh	2009-07-23 19:19:57.000000000 +0100
+++ hunspell-1.2.8/tests/test.sh	2009-07-23 19:20:45.000000000 +0100
@@ -33,8 +33,8 @@
 
 shopt -s expand_aliases
 
-alias hunspell='../src/tools/hunspell'
-alias analyze='../src/tools/analyze'
+alias hunspell='../libtool --mode=execute -dlopen ../src/hunspell/.libs/libhunspell*.la ../src/tools/hunspell'
+alias analyze='../libtool --mode=execute -dlopen ../src/hunspell/.libs/libhunspell*.la ../src/tools/analyze'
 
 if [ "$VALGRIND" != "" ]; then
   rm -f $TEMPDIR/test.pid*
